fiftyone/core/dataset.py

```diff
@@ -10133,6 +10081,10 @@ def _merge_samples_pipeline(
     #
     # Prepare frames merge pipeline
     #
+    # @todo this implementation creates a unique index on a temporary
```

Comment:
Do we have a ticket for this so it's not just lost in code?

Reply:
It's been resolved in https://github.com/voxel51/fiftyone/pull/6370!